### PR TITLE
[DICP] Implement paras auto cast in some binary ops.

### PR DESCRIPTION
**Pull request by Reinerzhou**
Implement paras auto cast in some binary ops.

### CODE REVIEW

**Review by jinminxi104 — APPROVED** [review]

File: dicp/dicp/vendor/TopsGraph/conversion.py [hunk 1]
```diff
@@ -110,6 +155,16 @@ def Mul(self, a, b):
             if hasattr(a.node, "meta") and 'val' in a.node.meta:
                 if (a.node.meta['val'].dtype == torch.complex64) or (a.node.meta['val'].dtype == torch.cfloat):
                     return tops_op.ComplexMul(a, b)
+        if isinstance(a, Proxy) and isinstance(b, Proxy):
```
Comment (CyCle1024) [inline dicp/dicp/vendor/TopsGraph/conversion.py:158]: 建议整块重复逻辑也封装为一个函数，binary dtype cast

**Review by CyCle1024 — CHANGES_REQUESTED** [review]

File: dicp/dicp/vendor/TopsGraph/conversion.py [hunk 2]
```diff
@@ -72,6 +73,50 @@ def register_conversion(aten_fn):
     )
 
 
+def get_cast_dtype(
```
Comment (jinminxi104) [inline dicp/dicp/vendor/TopsGraph/conversion.py:76]: 这块也和华为的整合一下，找个common的地方放一下

**Review by jinminxi104 — CHANGES_REQUESTED** [review]

File: dicp/dicp/vendor/TopsGraph/conversion.py (hunk 1 above)
Comment (Reinerzhou) [reply dicp/dicp/vendor/TopsGraph/conversion.py:125]: done.

File: dicp/dicp/vendor/TopsGraph/conversion.py (hunk 2 above)
Comment (Reinerzhou) [reply dicp/dicp/vendor/TopsGraph/conversion.py]: done.

**Review by CyCle1024 — APPROVED** [review]

**Review by jinminxi104 — APPROVED** [review]

**Pull request merged**